plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -2072,6 +2121,17 @@ protected StartAnswer execute(StartCommand cmd) {
     }
 
     /**
+     * Update boot args with the new nic mac addresses.
+     */
+    protected String replaceNicsMacSequenceInBootArgs(String oldMacSequence, String newMacSequence, VirtualMachineTO vmSpec) {
+        String bootArgs = vmSpec.getBootArgs();
+        if (!StringUtils.isEmpty(bootArgs) && !StringUtils.isEmpty(oldMacSequence) && !StringUtils.isEmpty(newMacSequence)) {
+            return bootArgs.replace(oldMacSequence, newMacSequence);
+        }
+        return "";
```

Comment:
@rafaelweingartner  May not be possible. `oldMacSequence` and `newMacSequence` are generated from the NICs info. Both would be empty only when no NICs are specified for the VM, which will never happen. `bootArgs` would hold more info including the nics sequence.